cloud/src/autogluon/cloud/predictor/predictor.py

```diff
@@ -897,3 +951,125 @@ def _get_local_predictor_cls(self):
+    def predictor_type(self):
+        return 'image'
+
+    @property
+    def _realtime_predictor_cls(self):
+        return AutoGluonImageRealtimePredictor
+
+    def _get_local_predictor_cls(self):
+        from autogluon.vision import ImagePredictor
+        predictor_cls = ImagePredictor
+        return predictor_cls
+
+    def fit(
+        self,
+        predictor_init_args,
+        predictor_fit_args,
+        image_path,
+        **kwargs
+    ):
+        super().fit(
+            predictor_init_args,
+            predictor_fit_args,
+            image_path=image_path,
+            **kwargs
+        )
+
+    def predict_real_time(self, test_data, test_data_image_column=None, accept='application/x-parquet'):
+        """
+        Predict with the deployed SageMaker endpoint. A deployed SageMaker endpoint is required.
+        This is intended to provide a low latency inference.
+        If you want to inference on a large dataset, use `predict()` instead.
+
+        Parameters
+        ----------
+        test_data: Union(str, pandas.DataFrame)
+            The test data to be inferenced.
+            Can be a pandas.DataFrame, a numpy.ndarray, a local path or a s3 path to a csv file containing paths of test images.
+            Or a local path to a single image file.
+            Or a list of local paths to image files.
+        test_data_image_column: Optional(str)
+            If provided a pandas.DataFrame as the test_data, you must specify the column name corresponding to image paths.
+            Images has to live in the same directory specified by the column.
+        accept: str, default = application/x-parquet
+            Type of accept output content.
+            Valid options are application/x-parquet, text/csv, application/json
+
+        Returns
+        -------
+        Pandas.DataFrame
+        Predict results in DataFrame
+        """
+        assert self.endpoint, 'Please call `deploy()` to deploy an endpoint first.'
+        assert accept in VALID_ACCEPT, f'Invalid accept type. Options are {VALID_ACCEPT}.'
+
+        import cv2
+        import numpy as np
+
+        if isinstance(test_data, str):
+            image = cv2.imread(test_data)
+            if image is None:  # not an image
+                test_data = load_pd.load(test_data)
+            else:
+                test_data = image
+        if isinstance(test_data, list):
+            images = []
+            for image in test_data:
+                images.append(cv2.imread(image))
+            test_data = np.array(images, dtype=object)
+        if isinstance(test_data, pd.DataFrame):
+            assert test_data_image_column is not None, 'Please specify an image column name'
+            assert test_data_image_column in test_data, 'Please specify a valid image column name'
+
+            # Convert test data to be numpy array for network transfer
+            test_data = test_data[test_data_image_column].apply(lambda path: cv2.imread(path)).to_numpy()
```

Comment:
`.apply` is a very slow operation, consider using list comprehension + .zip() which is faster.

https://stackoverflow.com/questions/52607864/dataframe-list-comprehension-zip-loop-through-chosen-df-columns-efficien